### PR TITLE
Refactor for OCP 4.7

### PR DESCRIPTION
**Pull request by simu**
- [x] Deploy cluster in private network
- [x] Deploy LB pair using Terraform
- [x] Update ignition template for RHCOS 4.7
- [x] Introduce and use node-group submodule

### CODE REVIEW

File: security_groups.tf [hunk 1]
```diff
@@ -57,6 +72,38 @@ resource "exoscale_security_group_rules" "all_machines" {
...
+    protocol    = "TCP"
+    ports       = ["80", "443"]
+    cidr_list   = ["0.0.0.0/0", "::/0"]
+  }
+  ingress {
+    description = "Ingress controller UDP"
+    protocol    = "UDP"
+    ports       = ["80", "443"]
+    cidr_list   = ["0.0.0.0/0", "::/0"]
+  }
```
Comment (ccremer) [inline security_groups.tf:98]: Why also UDP on 80/443?

**Review by ccremer — APPROVED** [review]
Looking good!

Just a question though

File: security_groups.tf (hunk 1 above)
Comment (simu) [reply security_groups.tf:98]: I kept this rule from the previous version of the code, but I don't think it's needed looking at the [OCP4.7 documentation](https://docs.openshift.com/container-platform/4.7/installing/installing_bare_metal/installing-bare-metal.html#installation-network-user-infra_installing-bare-metal) again.

Comment (ccremer) [reply security_groups.tf:98]: ok. Might be worth adding a comment to this documentation in the source?

Comment (simu) [reply security_groups.tf:98]: The documentation is linked at the top of the file

**Review by ccremer — APPROVED** [review]

**Review by ccremer — APPROVED** [review]
I don't really understand every detail, but haven't seen major issues

File: variables.tf [hunk 2]
```diff
@@ -11,17 +12,28 @@ variable "rhcos_template" {
 }
 
 variable "base_domain" {
-  default = "ocp4-poc.appuio-beta.ch"
+  default = "appuio-beta.ch"
```
[the remark this answers is not in the document]
Comment (simu) [reply variables.tf:15]: Good point, I even made a note about making this mandatory at some point

**Pull request merged**